src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -233,6 +234,66 @@ public OrganisationsDetailResponse retrieveOrganisations() {
         return updatedOrganisationDetails;
     }
 
+    @Override
+    public OrganisationsDetailResponse retrieveOrganisations(String group) {
+        List<Organisation> retrievedOrganisations = organisationRepository.findAll();
+
+        List<Organisation> pendingOrganisations = new ArrayList<>();// = organisationRepository.findByStatus(OrganisationStatus.PENDING.name());
+        List<Organisation> activeOrganisations = new ArrayList<>();// = retrieveActiveOrganisationDetails();
+        List<Organisation> resultingOrganisations = new ArrayList<>();
+
+        retrievedOrganisations.forEach(organisation -> {
+            if (organisation.getStatus() == OrganisationStatus.ACTIVE) {
+                activeOrganisations.add(organisation);
+            }
+        });
+        retrievedOrganisations.forEach(organisation -> {
+            if (organisation.getStatus() == OrganisationStatus.PENDING) {
+                pendingOrganisations.add(organisation);
+            }
+        });
+
+        if (pendingOrganisations.isEmpty() && activeOrganisations.isEmpty()) {
+
+            log.info("No Organisations Retrieved...");
+            throw new EmptyResultDataAccessException(1);
+        }
+
+        List<Organisation> updatedOrganisationDetails = new ArrayList<>();
+        Map<String, Organisation> activeOrganisationDtls = new ConcurrentHashMap<>();
+
+        activeOrganisations.forEach(organisation -> {
+            if (!organisation.getUsers().isEmpty() && null != organisation.getUsers().get(0).getUserIdentifier()) {
+                activeOrganisationDtls.put(organisation.getUsers().get(0).getUserIdentifier(), organisation);
+            }
+        });
+
+        if (!CollectionUtils.isEmpty(activeOrganisations)) {
+
+            RetrieveUserProfilesRequest retrieveUserProfilesRequest = new RetrieveUserProfilesRequest(activeOrganisationDtls.keySet().stream().sorted().collect(Collectors.toList()));
+            updatedOrganisationDetails = RefDataUtil.getMultipleUserProfilesFromUp(userProfileFeignClient, retrieveUserProfilesRequest,
+                    "false", activeOrganisationDtls);
+
+        }
```

Comment:
I may have missed this in the changes.. but how are you getting the Contact information and PBA accounts for respective organisations?

Reply:
This is done when an OrganisationEntityResponse is created.